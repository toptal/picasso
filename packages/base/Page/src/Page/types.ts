export type ViewportWidthType = 'wide' | 'full'

export type ViewportWidth = {
  fullWidth?: boolean
  width?: ViewportWidthType
}

export type PageContextProps = ViewportWidth & {
  hasSidebar: boolean
  setHasSidebar: (value: boolean) => void
  hasTopBar: boolean
  setHasTopBar: (value: boolean) => void
}
