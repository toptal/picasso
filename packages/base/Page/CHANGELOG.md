# @toptal/picasso-page

## 1.0.9

### Patch Changes

- [#4224](https://github.com/toptal/picasso/pull/4224) [`c05387de5636ed094365d1eff67b955d84b81c61`](https://github.com/toptal/picasso/commit/c05387de5636ed094365d1eff67b955d84b81c61) Thanks [@TomasSlama](https://github.com/TomasSlama)!
- revert types for compound components

- Updated dependencies [[`c05387d`](https://github.com/toptal/picasso/commit/c05387de5636ed094365d1eff67b955d84b81c61)]:
  - @toptal/picasso-notification@1.0.6
  - @toptal/picasso-accordion@1.0.6
  - @toptal/picasso-avatar@1.0.4
  - @toptal/picasso-button@1.0.6
  - @toptal/picasso-menu@1.0.6
  - @toptal/picasso-tag@1.0.4
  - @toptal/picasso-user-badge@1.0.4
  - @toptal/picasso-autocomplete@1.0.7

## 1.0.8

### Patch Changes

- [#4216](https://github.com/toptal/picasso/pull/4216) [`3512588b06c3660471a68500275321c640278cf0`](https://github.com/toptal/picasso/commit/3512588b06c3660471a68500275321c640278cf0) Thanks [@TomasSlama](https://github.com/TomasSlama)!
- update types for the compound component

- Updated dependencies [[`3512588`](https://github.com/toptal/picasso/commit/3512588b06c3660471a68500275321c640278cf0)]:
  - @toptal/picasso-notification@1.0.5
  - @toptal/picasso-accordion@1.0.5
  - @toptal/picasso-avatar@1.0.3
  - @toptal/picasso-button@1.0.5
  - @toptal/picasso-menu@1.0.5
  - @toptal/picasso-tag@1.0.3
  - @toptal/picasso-user-badge@1.0.3
  - @toptal/picasso-autocomplete@1.0.6

## 1.0.7

### Patch Changes

- Updated dependencies [[`977da66`](https://github.com/toptal/picasso/commit/977da669eaa4ee5aefbe2acda773e3621e5981c4)]:
  - @toptal/picasso-tooltip@1.1.0

## 1.0.6

### Patch Changes

- [#4204](https://github.com/toptal/picasso/pull/4204) [`ae69e0e985c48542e7cc1df401d82ca91aa474ad`](https://github.com/toptal/picasso/commit/ae69e0e985c48542e7cc1df401d82ca91aa474ad) Thanks [@TomasSlama](https://github.com/TomasSlama)!
- fix Paper styles of the Hamburger menu on smaller screens
  after we changed the order of styles

## 1.0.5

### Patch Changes

- Updated dependencies [[`a1d523092cc4de6cb376156435b99b1e483f39b9`](https://github.com/toptal/picasso/commit/a1d523092cc4de6cb376156435b99b1e483f39b9)]:
  - @toptal/picasso-shared@14.0.1
  - @toptal/picasso-autocomplete@1.0.5
  - @toptal/picasso-avatar@1.0.2
  - @toptal/picasso-badge@1.0.2
  - @toptal/picasso-button@1.0.4
  - @toptal/picasso-menu@1.0.4
  - @toptal/picasso-notification@1.0.4
  - @toptal/picasso-tag@1.0.2
  - @toptal/picasso-tooltip@1.0.2
  - @toptal/picasso-utils@1.0.2
  - @toptal/picasso-user-badge@1.0.2
  - @toptal/picasso-accordion@1.0.4
  - @toptal/picasso-dropdown@1.0.3
  - @toptal/picasso-container@1.0.2
  - @toptal/picasso-typography@1.0.2
  - @toptal/picasso-icons@1.0.2
  - @toptal/picasso-logo@1.0.2

## 1.0.4

### Patch Changes

- Updated dependencies [[`d22bf5d913428b586e2813cf530f4e96d46b622e`](https://github.com/toptal/picasso/commit/d22bf5d913428b586e2813cf530f4e96d46b622e), [`d22bf5d913428b586e2813cf530f4e96d46b622e`](https://github.com/toptal/picasso/commit/d22bf5d913428b586e2813cf530f4e96d46b622e)]:
  - @toptal/picasso-dropdown@1.0.2
  - @toptal/picasso-menu@1.0.3
  - @toptal/picasso-button@1.0.3
  - @toptal/picasso-autocomplete@1.0.4
  - @toptal/picasso-accordion@1.0.3
  - @toptal/picasso-notification@1.0.3

## 1.0.3

### Patch Changes

- Updated dependencies [[`372e152aadc737cfacdf7f35110bc1ce2f35a90f`](https://github.com/toptal/picasso/commit/372e152aadc737cfacdf7f35110bc1ce2f35a90f)]:
  - @toptal/picasso-menu@1.0.2
  - @toptal/picasso-autocomplete@1.0.3

## 1.0.2

### Patch Changes

- Updated dependencies []:
  - @toptal/picasso-autocomplete@1.0.2
  - @toptal/picasso-button@1.0.2
  - @toptal/picasso-accordion@1.0.2
  - @toptal/picasso-notification@1.0.2

## 1.0.1

### Patch Changes

- [#4164](https://github.com/toptal/picasso/pull/4164) [`84d579cee65c88aeb6ba58318c43d8deb5201b30`](https://github.com/toptal/picasso/commit/84d579cee65c88aeb6ba58318c43d8deb5201b30) Thanks [@mkrl](https://github.com/mkrl)!
- migrate to Picasso by Parts
  - picasso is now distributed as a set of independent packages with the main package `@toptal/picasso` now being a collection of re-exported packages
- Updated dependencies [[`84d579cee65c88aeb6ba58318c43d8deb5201b30`](https://github.com/toptal/picasso/commit/84d579cee65c88aeb6ba58318c43d8deb5201b30)]:
  - @toptal/picasso-shared@14.0.0
  - @toptal/picasso-autocomplete@1.0.1
  - @toptal/picasso-notification@1.0.1
  - @toptal/picasso-typography@1.0.1
  - @toptal/picasso-accordion@1.0.1
  - @toptal/picasso-container@1.0.1
  - @toptal/picasso-user-badge@1.0.1
  - @toptal/picasso-dropdown@1.0.1
  - @toptal/picasso-tooltip@1.0.1
  - @toptal/picasso-avatar@1.0.1
  - @toptal/picasso-button@1.0.1
  - @toptal/picasso-badge@1.0.1
  - @toptal/picasso-icons@1.0.1
  - @toptal/picasso-utils@1.0.1
  - @toptal/picasso-logo@1.0.1
  - @toptal/picasso-menu@1.0.1
  - @toptal/picasso-tag@1.0.1
