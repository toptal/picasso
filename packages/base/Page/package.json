{
  "name": "@toptal/picasso-page",
  "version": "2.0.8",
  "description": "Toptal UI components library - Page",
  "publishConfig": {
    "access": "public"
  },
  "main": "./dist-package/src/index.js",
  "module": "./dist-package/src/index.js",
  "scripts": {
    "build:package": "tsc -b tsconfig.json",
    "prepublishOnly": "yarn build:package"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/toptal/picasso.git"
  },
  "author": "Toptal",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/toptal/picasso/issues"
  },
  "homepage": "https://github.com/toptal/picasso/tree/master/packages/picasso#readme",
  "dependencies": {
    "@toptal/picasso-accordion": "1.0.15",
    "@toptal/picasso-autocomplete": "1.0.16",
    "@toptal/picasso-avatar": "2.0.2",
    "@toptal/picasso-badge": "1.0.3",
    "@toptal/picasso-button": "2.0.1",
    "@toptal/picasso-container": "1.0.3",
    "@toptal/picasso-dropdown": "2.0.3",
    "@toptal/picasso-icons": "1.1.1",
    "@toptal/picasso-logo": "1.0.4",
    "@toptal/picasso-menu": "1.0.11",
    "@toptal/picasso-notification": "2.0.7",
    "@toptal/picasso-shared": "15.0.0",
    "@toptal/picasso-tag": "1.1.2",
    "@toptal/picasso-tooltip": "1.1.2",
    "@toptal/picasso-typography": "2.0.1",
    "@toptal/picasso-user-badge": "2.0.2",
    "@toptal/picasso-utils": "1.0.3",
    "classnames": "^2.5.1",
    "tailwind-merge": "^2.2.2"
  },
  "sideEffects": [
    "**/styles.ts",
    "**/styles.js"
  ],
  "peerDependencies": {
    "@material-ui/core": "4.12.4",
    "@toptal/picasso-provider": "*",
    "react-dom": "^18.2.0",
    "@toptal/picasso-tailwind": "^2.5.0",
    "react": ">=16.12.0 < 19.0.0"
  },
  "devDependencies": {
    "@toptal/picasso-provider": "4.2.1",
    "@toptal/picasso-test-utils": "1.1.1",
    "react-dom": "^18.2.0",
    "styled-components": "^6.1.1"
  },
  "exports": {
    ".": "./dist-package/src/index.js"
  },
  "files": [
    "dist-package/**",
    "!dist-package/tsconfig.tsbuildinfo",
    "src"
  ]
}
