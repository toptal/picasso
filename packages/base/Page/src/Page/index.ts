import type { OmitInternalProps } from '@toptal/picasso-shared'

import type { Props } from './Page'

export { default, PageContext } from './Page'
export type { PageContextProps } from './types'
export type PageProps = OmitInternalProps<Props>
