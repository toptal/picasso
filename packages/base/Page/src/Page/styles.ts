import type { Theme } from '@material-ui/core/styles'
import { createStyles } from '@material-ui/core/styles'

export default ({ layout, palette }: Theme) =>
  createStyles({
    root: {
      display: 'flex',
      flexDirection: 'column',
      height: '100%',
      background: palette.grey.lightest,
      ...(layout.contentMinWidth && { minWidth: layout.contentMinWidth }),

      '& > footer, & > header': {
        flex: 0,
      },
      '& > header + *': {
        marginTop: 'var(--header-height)',
      },
    },
  })
