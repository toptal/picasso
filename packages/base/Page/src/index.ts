export * from './Page'
export * from './PageArticle'
export * from './PageAutocomplete'
export * from './PageCompound'
export * from './PageContent'
export * from './PageFooter'
export * from './PageHelmet'
export * from './PageBanner'
export * from './PageSidebar'
export * from './SidebarLogo'
export * from './SidebarItem'
export * from './SidebarMenu'
export * from './SidebarItemContent'
export * from './PageTopBar'
export * from './PageTopBarCompound'
export * from './PageTopBarMenu'
export * from './TopBarItem'
export * from './TopBarMenu'
export * from './PageHamburger'
export * from './PageHead'
export * from './PageHeadBase'
