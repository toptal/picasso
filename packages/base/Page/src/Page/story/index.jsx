import pageHelmetStory from '../../PageHelmet/story'
import pageContentStory from '../../PageContent/story'
import pageFooterStory from '../../PageFooter/story'
import pageBannerStory from '../../PageBanner/story'
import pageAutocompleteStory from '../../PageAutocomplete/story'
import pageArticleStory from '../../PageArticle/story'
import { Page } from '../Page'
import PicassoBook from '~/.storybook/components/PicassoBook'

const page = PicassoBook.section('Layout').createPage(
  'Page',
  `A Page component

  ${PicassoBook.createSourceLink(__filename)}
  `
)

page
  .createTabChapter('Props')
  .addComponentDocs({ component: Page, name: 'Page' })
  .addComponentDocs(pageHelmetStory.componentDocs)
  .addComponentDocs(pageContentStory.componentDocs)
  .addComponentDocs(pageArticleStory.componentDocs)
  .addComponentDocs(pageFooterStory.componentDocs)
  .addComponentDocs(pageBannerStory.componentDocs)
  .addComponentDocs(pageAutocompleteStory.componentDocs)

page
  .createChapter()
  .addExample(
    'Page/story/Default.example.tsx',
    {
      title: 'Default',
      description:
        'Page has centered content and restricted width of 1200px or 75rem',
    },
    'base/Page'
  )
  .addExample(
    'Page/story/WideWidth.example.tsx',
    {
      title: 'Wide width',
      description:
        'Page has centered content and restricted width of 1440px or 90rem',
    },
    'base/Page'
  )
  .addExample(
    'Page/story/FullWidth.example.tsx',
    {
      title: 'Full width',
    },
    'base/Page'
  )
  .addExample(
    'Page/story/Scroll.example.tsx',
    'Scroll with overflow',
    'base/Page'
  )
  .addExample(
    'Page/story/WithBanner.example.tsx',
    {
      title: 'With Banner',
    },
    'base/Page'
  )
  .addExample(
    'Page/story/WithCompoundBanner.example.tsx',
    {
      title: 'With Compound Banner',
    },
    'base/Page'
  )
  .addExample(
    'Page/story/WithBannerAndSidebar.example.tsx',
    {
      title: 'With Banner and Sidebar',
    },
    'base/Page'
  )

page.connect(pageHelmetStory.chapter)

page.connect(pageContentStory.chapter)

page.connect(pageArticleStory.chapter)

page.connect(pageFooterStory.chapter)

page.connect(pageBannerStory.chapter)

page.connect(pageAutocompleteStory.chapter)
