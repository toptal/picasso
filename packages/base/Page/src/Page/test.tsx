import React from 'react'
import type { RenderResult } from '@toptal/picasso/test-utils'
import { render } from '@toptal/picasso/test-utils'

import Page from './Page'

const renderPage = (children: React.ReactNode) => {
  return render(<Page>{children}</Page>)
}

describe('Page', () => {
  let api: RenderResult

  beforeEach(() => {
    api = renderPage(<div>Test</div>)
  })
  it('renders', () => {
    const { container } = api

    expect(container).toMatchSnapshot()
  })
})
