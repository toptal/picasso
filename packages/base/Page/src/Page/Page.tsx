import type { ReactNode, HTMLAttributes } from 'react'
import React, { forwardRef } from 'react'
import type { Theme } from '@material-ui/core/styles'
import { makeStyles } from '@material-ui/core/styles'
import cx from 'classnames'
import type { BaseProps } from '@toptal/picasso-shared'

import { PageHamburgerContextProvider } from '../PageHamburger'
import type { PageContextProps, ViewportWidthType } from './types'
import styles from './styles'

export interface Props extends BaseProps, HTMLAttributes<HTMLDivElement> {
  /** DEPRECATED! Component becomes responsive with width 100% and overrides width prop */
  fullWidth?: boolean
  /** Define container width `wide` | `full` */
  width?: ViewportWidthType
  /** Horizontally centers the content */
  centered?: boolean
  /** Children components (`Page.TopBar`, `Page.Content`, `Page.Footer`) */
  children: ReactNode
  hamburgerId?: string
}

export const PageContext = React.createContext<PageContextProps>({})

const useStyles = makeStyles<Theme>(styles, {
  name: 'Page',
})

// eslint-disable-next-line react/display-name
export const Page = forwardRef<HTMLDivElement, Props>(function Page(
  props,
  ref
) {
  const {
    children,
    className,
    hamburgerId = 'hamburger',
    style,
    width,
    fullWidth,
    ...rest
  } = props
  const classes = useStyles()

  return (
    <div
      {...rest}
      ref={ref}
      className={cx(classes.root, className)}
      style={{ ...style, '--header-height': '3.5rem' } as React.CSSProperties}
    >
      <PageContext.Provider value={{ width, fullWidth }}>
        <PageHamburgerContextProvider hamburgerId={hamburgerId}>
          {children}
        </PageHamburgerContextProvider>
      </PageContext.Provider>
    </div>
  )
})

Page.displayName = 'Page'

export default Page
