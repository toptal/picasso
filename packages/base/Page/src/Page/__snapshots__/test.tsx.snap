// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders 1`] = `
<div>
  <div
    class="Picasso-root"
  >
    <div
      class="Page-root"
      style="--header-height: 3.5rem;"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</div>
`;
